python_ta/transforms/setendings.py

```diff
@@ -107,24 +100,97 @@
     astroid.YieldFrom
 ]
 
-
-# Helpers for setting locations based on source code.
-def _is_close_paren(s, index):
-    return s[index] == ')'
-
-
-def _is_open_paren(s, index):
-    return s[index] == '('
+# Predicate functions, for setting locations based on source code.
+# Predicates can only return a single truthy value, because of how its used in
+# `astroid/transforms.py`
+# ====================================================
+def _token_search(token):
+    """
+    @type token: string
+    @rtype: function
+    """
+    def _is_token(s, index, node):
+        """Fix to include certain tokens such as a paren, bracket, or brace.
+        @type s: string
+        @type index: int
+        @type node: Astroid node
+        @rtype: bool
+        """
+        return s[index] == token
+    return _is_token
+
+def _keyword_search(keyword):
+    """
+    @type keyword: string
+    @rtype: function
+    """
+    def _is_keyword(s, index, node):
+        """Search for a keyword. Right-to-left.
+        @type s: string
+        @type index: int
+        @type node: Astroid node
+        @rtype: bool
+        """
+        return s[index : index + len(keyword)] == keyword
+    return _is_keyword
+
+def _is_within_close_bracket(s, index, node):
+    """Fix to include right ']'."""
+    # print('>>>>>>', s, s[index], index, node.end_col_offset)
+    if index >= len(s)-1: return False
```

Comment:
I believe I have already asked you to stop using one-line ifs. Please change this and all the others.